### ISSUE TITLE
Hosting RoslynCodeEditor in a WPF dialog

### ISSUE THREAD

**Issue by nguyencongle** [aelij/RoslynPad#337]
Hi,
I can host the RoslynCodeEditor in our own WPF dialog, and everything is working correctly.
But everytime I open the dialog, the memory keep increasing by 200MB.
I tried to dispose the CompositionHost when close the dialog, but it just release a little bit of memory.
Is there any effective way to do that?
Thank you.

**Comment by aelij** [comment]
Hi @nguyencongle. Unfortunately I do not have time to investigate this at the moment. I suggest you use a memory profile and see what holds the object graph. If you find a leak, please report it back and I'll fix it. Thanks 🙏🏽

**Comment by inf9144** [comment]
Wpf likes too keep it's memory and does not want to give it back until it's needed by OS or another app. You can force it to release the memory - dont ask me why this codeline works - but it does: 

System.Diagnostics.Process.GetCurrentProcess().MinWorkingSet = System.Diagnostics.Process.GetCurrentProcess().MinWorkingSet;

It's from one of the last posts here - and i do make highly use of it everytime when any of my windows in wpf projects are closed:

https://social.msdn.microsoft.com/Forums/vstudio/en-US/7d470f2b-d836-4044-b457-dfa4221b189f/how-to-analyze-reason-for-extreme-high-part-of-unmanaged-memory-in-c-wpf-application?forum=wpf

**Comment by nguyencongle** [comment]
OOPS, this line of code is so excited. when I open the dialog it goes up to 400MB, with this line when I close the dialog it goes down to just 30MB. I am wondering how can it be, and why WPF keeps a lot of garbage instances like this.
Thanks a lot @inf9144 .

**Comment by trrahul** [comment]
I suspect this trick only momentarily reduces the size of the physical memory reserved for the process. When I tested it, the memory usage reduced and I noticed a hit in performance. I also noticed increased hard disk activity and I could be wrong but I think this statement forces the excess primary memory to write to the pagefile and the unexpected page faults were the reason for the momentary degraded performance.
